### PR TITLE
Our very own templating engine for utterances.

### PR DESCRIPTION
**Pull request by vqtran**
So there's this existing templating engine called alexa-utterances. Its
the only one I can find so far. I tried using it for a while but
realized that:

1. It has a lot of necessary features for us.
2. Has an ugly syntax.
3. Most legit, its syntax plays poorly with custom slot types.
4. Their code is a mess and would be a pain to edit.

So armed with a six pack of Lagunitas Little Sumpin' Sumpin' I set out
to roll my own templating engine for these utterances. It's pretty damn
simple so far, but gets the job done much better than alexa-utterance.
Here's the only two things this thing does:

  [a|b|c] - Definite variation. An utterance will be made for each version
            listed here.

  Ex: "Hi [Gabe|Vinh Tran]" => ["Hi Gabe", "Hi Vinh Tran"]

  (a|b|c) - Optional variation. An utterance will be made for each version
            listed here, as well as one without.

  Ex: "Hi (Gabe|Vinh Tran)" => ["Hi", "Hi Gabe", "Hi Vinh Tran"]

I then rewrote our sample utterances into a much much shorter template!
With this now making conditionals will be a lot more reasonable.

### CODE REVIEW

**Comment by GabeLoins** [comment]
Couldn't you simplify the [] / () system by implementing () in []?

e.g.

('Gabe', 'Vinh') -> ['', 'Gabe', 'Vinh']?

**Comment by vqtran** [comment]
Yeah that's good, I'll update that when i get a chance

File: speech-assets/UtterancesTemplate.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+HelpIntent help (me)
+HelpIntent get help
+HelpIntent what [can|do] you [do|tell me]
+HelpIntent how [can|do] I use [you|this]
+HelpIntent what (commands|questions|things|stuff) can I \
+  [ask|ask you|say|do|use|use this for]
+
+CountIntent How [many|much] (entries in|rows in) {TableName} \
+  (are there|do we have)
+CountIntent [What's|What is|Count|Find] the [number|size|length|cardinality] \
```
Comment (GabeLoins) [inline speech-assets/UtterancesTemplate.txt]: brilliant!

**Comment by GabeLoins** [comment]
Should we just merge for now?

**Comment by vqtran** [comment]
Yeah why not, we'll probably end up including the change you mentioned in future things anyways as we'll find more problems with this

**Pull request merged**